### PR TITLE
fix: [PL-41518]: fix project/org resource group issues

### PR DESCRIPTION
**Pull request by namanharness**
### Summary

This PR is to incorporate Organization and Project as a RG and fix UI associated bugs.
Few changes made specifically in UI:
- in Account Scope, Project will only be shown for non-current ResourceGroup scopes
- On the Left, in ResourceTypeList: Remove Projects from the list when user is in Account scope and has chosen "CURRENT" scope from the dropdown & Category is `ADMINSTRATIVE_FUNCTIONS`
- On the Right, in Resource Cards, "Specified" option will only be shown when user has chosen "CURRENT" scope from the dropdown
- Added a new column to show Orgs in `ProjectResourceModalBody`
- Added `Orgidentifier` useGetProjectList call 

https://github.com/harness/harness-core-ui/assets/110456765/42a93926-ca66-48bf-b490-347933b8ad95

https://github.com/harness/harness-core-ui/assets/110456765/e7d38ed3-7822-46cc-9016-72de5d19198e

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/20-rbac/components/ResourceTypeList/ResourceTypeList.tsx [hunk 1]
```diff
@@ -49,6 +62,18 @@ const ResourceTypeList: React.FC<ResourceTypeListProps> = props => {
   const getIntermittent = (resourceTypes: ResourceType[]): boolean =>
     Array.from(resourceTypes).some(value => preSelectedResourceList.includes(value))
 
+  const getResourceTypes = (resourceCategory: ResourceType | ResourceCategory) => {
```
Comment (namanharness) [inline src/modules/20-rbac/components/ResourceTypeList/ResourceTypeList.tsx:65]: this handles the resources in the left resources bar where we show PROJECTS on conditions

File: src/modules/20-rbac/pages/ResourceGroupDetails/ResourceGroupDetails.tsx [hunk 2]
```diff
@@ -140,6 +141,15 @@ const ResourceGroupDetails: React.FC = () => {
     setResourceCategoryMap(_map => RbacFactory.getResourceCategoryList(types))
   }, [selectedScope, resourceTypeData, includedScopes])
 
+  useDeepCompareEffect(() => {
+    if (selectedResourcesMap) {
+      if (resourceGroupScope === Scope.ACCOUNT && selectedScope === SelectorScope.CURRENT) {
```
Comment (namanharness) [inline src/modules/20-rbac/pages/ResourceGroupDetails/ResourceGroupDetails.tsx:146]: this is to remove the resource entry from list in the main body

File: src/modules/20-rbac/pages/ResourceGroupDetails/ResourceGroupDetails.tsx [hunk 3]
```diff
@@ -140,6 +141,15 @@ const ResourceGroupDetails: React.FC = () => {
...
+        if (selectedResourcesMap.has(ResourceType.PROJECT)) {
+          selectedResourcesMap.delete(ResourceType.PROJECT)
```
Comment (mayankVermaHarness) [inline src/modules/20-rbac/pages/ResourceGroupDetails/ResourceGroupDetails.tsx:148]: use set state method: `setSelectedResourceMap`

Comment (namanharness) [reply src/modules/20-rbac/pages/ResourceGroupDetails/ResourceGroupDetails.tsx]: Done

**Review by mayankVermaHarness — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks good

**Pull request merged**